### DISCUSSION TITLE
[3.x]: getMimeType() should accept transforms as param and return the transformed asset mime type

### DISCUSSION THREAD

**Discussion by therealpecus** [craftcms/cms#12269] [Ideas] [answered]
### Description

The `getMimeType()` method of Assets should accept a transform as parameter and in this case return the transformed asset mime type, instead of the original asset mime type.

### Steps to reproduce

1. declare a transform in a template: `{% set hero = { mode='fit', width='1280', format='webp' } %}
2. request the mime type of the asset: `{{ asset.getMimeType(hero) }}`

### Expected behavior

The function should return the mime type for the transformed asset (i.e. `image/webp`).

### Actual behavior

The function does not accept parameters and returns the mime type for the original asset (assuming for this case `image/png`)

See https://github.com/craftcms/cms/blob/v3/src/elements/Asset.php#L1387

**Comment by brianjhanson** [comment]
@therealpecus I moved this over to discussions because this is working as expected for Craft 3, but we have put this functionality on the roadmap (aiming for 4.4 at the moment).

**Reply by therealpecus** [reply]
Thanks @brianjhanson, I take then that feature updates are not aimed at v3 anymore, and only security fixes?

**Reply by brianjhanson** [reply]
Yep! Bugfixes are still getting ported back to 3 as well, but new features are strictly 4.

**Comment by mylesthedev** [comment]
Just ran into this issue, would love to see a patch for 4.*

**Comment by brandonkelly** [comment, marked as answer]
This is in for the upcoming Craft 3.8 and 4.4 releases 🎉 (#12397)

**Reply by brandonkelly** [reply]
3.8 and 4.4 are out now!
